lib/amqp_client.js

```diff
@@ -215,17 +215,13 @@ AMQPClient.prototype.createSender = function(address, options) {
 };
 
 /**
- * Set up a callback to be called whenever message is received from the given source (subject to the given filter).
- * Callback is called with (error, message), and the payload is decoded using the receiver link policy's
- * decoder method. The method itself returns a promise which is resolved with receiver link information
- * once it's attached.
+ * Creates a receiver link for the given address, with optional link policy. The
+ * promise returned resolves to a link that is an EventEmitter, which can be
+ * used to listen for 'message' events.
  *
  * @method createReceiver
- * @param {string} [source]     Source of the link to connect to.  If not provided will use default queue from connection uri.
- * @param {Object} [options]    Options used for link creation
- * @param options.filter        Filter used in connecting to the source.  See AMQP spec for details, and your server's documentation
- *                              for possible values.  node-amqp-encoder'd maps will be translated, and simple maps will be converted
- *                              to AMQP Fields type as defined in the spec.
+ * @param {String} [source]     Source of the link to connect to.  If not provided will use default queue from connection uri.
+ * @param {Object} [options]    Link policy used for creating this receiver link
```

Comment:
It's just a start :) I figured it was better to have some code up on a branch to track the conversation.